src/core/stream/period/period_stream.ts

```diff
@@ -235,6 +251,9 @@ export default function PeriodStream(
         return askForMediaSourceReload(relativePosAfterSwitch,
                                        true,
                                        streamCanceller.signal);
+      } else if (strategy.type !== "continue" && relativePosAfterSwitch) {
+        const seekingTime = playbackObserver.getCurrentTime() + relativePosAfterSwitch;
+        callbacks.seekTo(seekingTime);
```

Comment:
Isn't the only other case the one corresponding to `"flush-buffer"`, in which case it could be integrated to needsBufferFlush.

Reply:
yes It's true, I could add it to the callback `needsBufferFlush` I would need to add a param for the seeking time